### DISCUSSION TITLE
Inline code chunks which depend on some libraries with 'cache: true' throw errors

### DISCUSSION THREAD

**Discussion by setseed13** [quarto-dev/quarto-cli#4434] [Q&A] [answered]
Hi!

Here is a minimal example of my current file:

`````
---
title: "minimal"
format: html
execute:
    cache: true
    echo: false
---
```{r}
library(tidyverse)
some_data <- cbind(c(1,2), c(3,4))
```

some text

```{r}
some_data %>% .[1,1]
```
`````
(Didn't know how to excape the backticks, sorry...edit: thanks, edited it)

This works. When I, however, add inline code after having rendered the file:
```
more text

`r some_data %>% .[1,1]`
```
I get the following error
```
processing file: minimal.qmd
  |....................................................| 100%                  Quitting from lines 19-22 (minimal.qmd) 
                                                                                                            Error in some_data %>% .[1, 1] : could not find function "%>%"
Calls: .main ... inline_exec -> hook_eval -> withVisible -> eval -> eval

Execution halted
```
Adding a chunk instead of inline code works, though.

Is there anything mangled up on my side?
Anyway to change that behavior?

**Comment by mcanouil** [comment, marked as answer]
It works just fine for me without and with the inline R code with knitr 1.42 and Quarto 1.2.335.
Cache is for variables not packages loading, so you need to be sure calls to `library` & co are not cached.

Your need more backticks around (I am using a block with five `` ` ``):

`````
````
---
title: "minimal"
format: html
execute:
  cache: true
  echo: false
---

```{r}
library(magrittr)
```

```{r}
some_data <- cbind(c(1,2), c(3,4))
```

some text

```{r}
some_data %>% head()
```

more text

`r some_data %>% head()`
````
`````

PS: why `some_data %>% .[1,1]`? Even for a demo about the issue of not having the function, it should never be written, in my opinion.

**Reply by setseed13** [reply]
> Cache is for variables not packages loading, so you need to be sure calls to `library` & co are not cached.

Didn't know that, but certainly should have tried.
Added that option to my library loading chunk, works as expected now, since it actually makes a lot of sense...
Thank you so much!

> PS: why some_data %>% .[1,1]?

My intention was to use a minimal example from a library a lot of people have. If this is about what it actual does: the first thing that came to my mind.
Is it a style problem? Then I have no idea, I'm just learning :)

**Reply by mcanouil** [reply]
> My intention was to use a minimal example from a library a lot of people have. If this is about what it actual does: the first thing that came to my mind.

I've got that ;)

> Is it a style problem? Then I have no idea, I'm just learning :)

Simply an opinion, my rule of thumb is "if there is only one pipe, there is one too many", i.e., the pipe is supposed to improve readability: `some_data %>% .[1, 1]` does not feel easier to read than `some_data[1, 1]`, same goes for `some_data %>% head()` compared to `head(some_data)`.

**Reply by setseed13** [reply]
Thanks for the input!
So far I just hacked my way through without any style considerations, I guess it's time to put some more thought into it :)

**Reply by mcanouil** [reply]
Here an opinionated guide: <https://style.tidyverse.org>.

The main and only thing important is to be consistent and kind to your future self 😉

**Reply by setseed13** [reply]
> The main and only thing important is to be consistent and kind to your future self 😉

So far, I achieved neither of these two things :)

Thanks for the link, pushed it to the top of my reading list!
